lab_3_ann_retriever/main.py

```diff
@@ -269,6 +378,26 @@ def retrieve_relevant_documents(
 
         In case of corrupt input arguments, None is returned.
         """
+        if (n_neighbours <= 0 or not isinstance(query, str)
+                or not query or not isinstance(n_neighbours, int)):
+            return None
+
+        query_vectorized = self._index_document(query)
+        if query_vectorized is None:
+            return None
+
+        self.index_documents(self._documents)
+        knn_list_of_tuples = self._calculate_knn(query_vectorized, self._document_vectors, n_neighbours)
+        if knn_list_of_tuples is None or not knn_list_of_tuples:
+            return None
+
+        relevant_docs = []
+        for index, float_as_value in knn_list_of_tuples:
+            if float_as_value is not None:
```

Comment:
```
if float_as_value is None:
    return None
```

Reply:
If I've understood right, I needed to add this to my code before my own check for foat_as_value. Done

Reply:
Vasilisa Mikhailovna means that you need to reverse your condition and exit from the function. I believe you need to remove this else now.